vars/pushDockerImages.groovy

```diff
@@ -0,0 +1,139 @@
+  )
+
+*/
+def call(Map args = [:]) {
+  if(!isUnix()){
+    error('publishToCDN: windows is not supported yet.')
+  }
+  def registry = args.containsKey('registry') ? args.registry : error('pushDockerImages: registry parameter is required')
+  def secret = args.containsKey('secret') ? args.secret : error('pushDockerImages: secret parameter is required')
+  def targetNamespace = args.containsKey('targetNamespace') ? args.targetNamespace : error('pushDockerImages: targetNamespace parameter is required')
+  def version = args.containsKey('version') ? args.version : error('pushDockerImages: version parameter is required')
+  def imageName = args.containsKey('imageName') ? args.imageName : error('pushDockerImages: imageName parameter is required')
+  def arch = args.get('arch', 'amd64')
+  def variants = args.get('variants', [:])
+  def snapshot = args.get('snapshot', true)
+
+  // Transform version in a snapshot.
+  def sourceTag = version
+  def aliasVersion = ""
+  if (snapshot) {
+    // remove third number in version
+    aliasVersion = version.substring(0, version.lastIndexOf(".")) + "-SNAPSHOT"
+    sourceTag += "-SNAPSHOT"
+  }
+
+  // What docker tags are gonna be used
+  def tags = calculateTags(sourceTag, aliasVersion)
+
+  println tags
+  dockerLogin(secret: "${secret}", registry: "${registry}")
+  variants?.each { variant, sourceNamespace ->
+    tags.each { tag ->
+      // TODO:
+      // For backward compatibility let's ensure we tag only for amd64, then E2E can benefit from until
+      // they support the versioning with the architecture
+      if ("${arch}" == "amd64") {
+        doTagAndPush(registry: registry, imageName: imageName, variant: variant, sourceTag: sourceTag, targetTag: "${tag}", sourceNamespace: sourceNamespace, targetNamespace: targetNamespace)
+      }
+      doTagAndPush(registry: registry, imageName: imageName, variant: variant, sourceTag: sourceTag, targetTag: "${tag}-${arch}", sourceNamespace: sourceNamespace, targetNamespace: targetNamespace)
+    }
+  }
+}
+
+/**
+* Tag and push the source docker image. It retries to add resilience.
+*
+* @param imageName of the docker image
+* @param variant name of the variant used to build the docker image name
+* @param sourceNamespace namespace to be used as source for the docker tag command
+* @param targetNamespace namespace to be used as target for the docker tag command
+* @param sourceTag tag to be used as source for the docker tag command, usually under the 'beats' namespace
+* @param targetTag tag to be used as target for the docker tag command, usually under the 'observability-ci' namespace
+*/
+def doTagAndPush(Map args = [:]) {
+  def name = args.imageName
+  def variant = args.variant
+  def sourceTag = args.sourceTag
+  def targetTag = args.targetTag
+  def sourceNamespace = args.sourceNamespace
+  def targetNamespace = args.targetNamespace
+  def registry = args.registry
+
+  def sourceName = "${registry}/${sourceNamespace}/${name}${variant}:${sourceTag}"
```

Comment:
Not sure is correct

```
def registry = 'docker.elastic.co'
def sourceNamespace = 'observability-ci'
def name = 'metricbeat'
variant = 'cloud'
sourceTag = '8.0.0-SNAPSHOT'

docker.elastic.co/observability-ci/metricbeatcloud:8.0.0-SNPASHOT
```

Reply:
I don't understand your question, `sourceNamespace` is never `observability-ci` but the one the consumer decides. That's exactly the same implementation as used to be -> https://github.com/elastic/beats/pull/30414/files#diff-5edb5ab93fc95960129eecb2ee3d8763a9c5e02b56e5b80126e0c5a52296ee8fL352

Reply:
aha! `variant` should contain `-cloud` rather than `cloud`

Reply:
the values are examples to point you to the separator (`-`) issue

Reply:
I don't think that's an issue:

`"${registry}/${sourceNamespace}/${name}${variant}:${sourceTag}"` then variable names should be defined with the format needed, otherwise what will you use?